include/pybind11/cast.h

```diff
@@ -655,8 +665,6 @@ class type_caster_generic {
         return inst.release();
     }
 
-protected:
```

Comment:
Yes. Direct access to `value` is needed in the `module_local_load` lambda. And seeing as `detail::type_caster_generic` is already an implementation detail, there isn't really much of a need to protect access.